tests/optimizers/test_dion_reference.py

```diff
@@ -0,0 +1,578 @@
+import pytest
```

Comment:
Can you also add a few tests for `dion.py` and `muon.py`? We expect people to use those instead of `dion_reference.py`